hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java

```diff
@@ -143,7 +143,11 @@ public void loadInstantDetailsInMemory(String startTs, String endTs) {
 
   public void loadCompletedInstantDetailsInMemory() {
     loadInstants(null, true,
-        record -> HoodieInstant.State.COMPLETED.toString().equals(record.get(ACTION_STATE).toString()));
+        record -> {
+          // Very old archived instants don't have action state set.
+          Object action = record.get(ACTION_STATE);
+          return action == null || HoodieInstant.State.COMPLETED.toString().equals(action.toString());
```

Comment:
When action equals null, the instant state is definite to be `COMPLETE` for old version ? Can we write ta test case?